pennylane/gradients/pulse_gradient.py

```diff
@@ -53,26 +54,43 @@ def split_evol_ops(op, word, word_wires, key):
         op (ParametrizedEvolution): operation to split up.
         word (str): Pauli word with respect to which to rotate between the split up parts.
         word_wires (~.wires.Wires): wires on which the Pauli word acts
-        key (list[int]): randomness seed to pass to the sampler for sampling the split-up point.
+        tau (float or tensor_like): split-up time(s). If multiple times are passed, the split-up
+            operations are set up to return intermediate time evolution results, leading to
+            broadcasting effectively.
 
     Returns:
-        float: The sampled split-up time tau
         tuple[list[`~.Operation`]]: The split-time evolution, expressed as three operations in the
             inner lists. The first tuple entry contains the operations with positive Pauli rotation
             angle, the second entry the operations with negative Pauli rotation angle.
     """
-    before_plus, after_plus, before_minus, after_minus = (copy(op) for _ in range(4))
-    # Extract time interval, split it, and set the intervals of the copies to the split intervals
     t0, t1 = op.t
-    tau = jax.random.uniform(key) * (t1 - t0) + t0
-    before_plus.t = before_minus.t = jax.numpy.array([t0, tau])
-    after_plus.t = after_minus.t = jax.numpy.array([tau, t1])
+    if broadcast := qml.math.ndim(tau) > 0:
```

Comment:
nice, learned something https://docs.python.org/3/whatsnew/3.8.html#assignment-expressions
I love that this is called a walrus operator, := will never look the same again 😆 

```suggestion
    if broadcast := (qml.math.ndim(tau) > 0):
```
for readability?

Reply:
Haha, yeah, they are amazing :) 

Unfortunately pylint complains about the parentheses... Could put an ignore statement though, whatever you prefer :)

Reply:
no strong preference